control/jet_vane_playground.cc

```diff
@@ -0,0 +1,195 @@
+// %deps(simple_geometry)
+// %deps(scene_tree)
+// %deps(window_3d)
+#include "third_party/experiments/viewer/primitives/scene_tree.hh"
+#include "third_party/experiments/viewer/primitives/simple_geometry.hh"
+#include "third_party/experiments/viewer/window_3d.hh"
+
+#include "control/jet_vane_mapper.hh"
+#include "control/jet_vane_model.hh"
+#include "control/quadraframe_model.hh"
+
+namespace jet {
+namespace control {
+
+namespace {
+void setup() {
+  const auto view = viewer::get_window3d("Mr. Vane, vanes");
+  view->set_target_from_world(SE3(SO3::exp(Eigen::Vector3d(-M_PI * 0.5, 0.0, 0.0)), Eigen::Vector3d::Zero()));
+  view->set_continue_time_ms(10);
+  view->set_zoom(10.0);
+
+  const auto background = view->add_primitive<viewer::SimpleGeometry>();
+  const geometry::shapes::Plane ground{jcc::Vec3::UnitZ(), 0.0};
+  background->add_plane({ground});
+  background->flip();
+}
+
+void put_plane(viewer::SimpleGeometry& geo, const SE3& world_from_plane) {
+  viewer::Polygon poly;
+
+  // The frame here looks like the following
+  //
+  // (X out of page)
+  // ------------------- -Z
+  // |
+  // |
+  // |
+  // |
+  // |
+  // |
+  // |
+  // +Y
+
+  constexpr double ONE_QUARTER = 1.0 / 4.0;
+  constexpr double THREE_QUARTERS = 3.0 / 4.0;
+
+  const jcc::Vec3 vane_tip_edge_from_cp(0.0, 0.0, 10e-3);
+
+  constexpr double vane_x = 20e-3;
+  constexpr double vane_z = 30e-3;
+
+  // These are all in the *angled* vane frame
+  const std::vector<jcc::Vec3> vane_points_vane_frame = {{
+      jcc::Vec3(-vane_x * ONE_QUARTER, 0.0, -vane_z),
+      jcc::Vec3(-vane_x * ONE_QUARTER, 0.0, 0.0),
+      jcc::Vec3(0.5 * vane_x, 0.0, 0.0),
+      jcc::Vec3(vane_x * THREE_QUARTERS, 0.0, -0.2 * vane_z),
+      jcc::Vec3(vane_x * THREE_QUARTERS, 0.0, -vane_z),
+  }};
+
+  for (const auto& pt_plane_frame : vane_points_vane_frame) {
+    poly.points.push_back(world_from_plane * (pt_plane_frame + vane_tip_edge_from_cp));
+  }
+  poly.outline = true;
+  poly.color = jcc::Vec4(0.8, 0.8, 0.8, 0.8);
```

Comment:
Unimportant -- should have a library of harmonious colors for improved code readability and ease of creating visualizations with well-differentiated objects.

Reply:
Going to remove this file